### PR TITLE
Removed all keycloak dependencies, replaced with nimbus jose

### PR DESCRIPTION
**Pull request by francis-pouatcha**

### CODE REVIEW

File: src/main/java/com/adorsys/ssi/sdjwt/SdJws.java [hunk 1]
```diff
@@ -1,122 +1,123 @@
...
 
-    public void verifySignature(SignatureVerifierContext verifier) throws VerificationException {
-        Objects.requireNonNull(verifier, "verifier must not be null");
-        try {
-            if (!verifier.verify(jwsInput.getEncodedSignatureInput().getBytes(StandardCharsets.UTF_8), jwsInput.getSignature())) {
-                throw new VerificationException("Invalid jws signature");
-            }
-        } catch (Exception e) {
-            throw new VerificationException(e);
+    public void verifySignature(JWSVerifier verifier) throws JOSEException, JOSEException {
```
Comment (Marcjazz) [inline src/main/java/com/adorsys/ssi/sdjwt/SdJws.java:89]: method does not throws JOSEException

**Review by Marcjazz — COMMENTED** [review]
I tested with jwt.io and it is working, I also did a local integration of the changes and everything is walking just fine. 

As for the code review, They are few unsued variables I addressed in comments

File: src/test/java/com/adorsys/ssi/sdjwt/TestSettings.java [hunk 2]
```diff
@@ -24,183 +25,69 @@ public class TestSettings {
...
+        return parseJwk(keySetting).toECKey(); // Where 'keyData' is your JWK
     }
 
-    private static KeyPair readKeyPair(JsonNode keySetting) {
-        String curveName = keySetting.get("crv").asText();
-        String base64UrlEncodedD = keySetting.get("d").asText();
-        String base64UrlEncodedX = keySetting.get("x").asText();
-        String base64UrlEncodedY = keySetting.get("y").asText();
-        return readEcdsaKeyPair(curveName, base64UrlEncodedD, base64UrlEncodedX, base64UrlEncodedY);
+    public static JWSVerifier verifierContextFrom(JsonNode keyData, String algorithm) throws ParseException, JOSEException {
```
[the remark this answers is not in the document]
Comment (francis-pouatcha) [reply src/test/java/com/adorsys/ssi/sdjwt/TestSettings.java:65]: Thanks. We need the algorithm. Added exception for other values.
Fixed with commit #3a6a88a

File: src/test/java/com/adorsys/ssi/sdjwt/TestSettings.java [hunk 3]
```diff
@@ -24,183 +25,69 @@ public class TestSettings {
...
-            String kid) {
-        JsonNode keySetting = keySettings.get(keyName);
-        KeyPair keyPair = readKeyPair(keySetting);
-        return getSignatureSignerContext(keyPair, algorithm, kid);
+        jwsAlgorithm = JWSAlgorithm.parse(testSettings.get("jwsAlgorithm").asText());
     }
 
-    private static SignatureVerifierContext initVerifierContext(JsonNode keySettings, String keyName, String algorithm,
-            String kid) {
+    private ECKey loadOrCreateECKey(JsonNode keySettings, String keyName) throws ParseException, JOSEException {
```
[the remark this answers is not in the document]
Comment (francis-pouatcha) [reply src/test/java/com/adorsys/ssi/sdjwt/TestSettings.java]: Fixed with [#3a6a88a](https://github.com/adorsys/sd-jwt/pull/3/commits/3a6a88a760dd18bff56bbcb487bcfada956378b1)

File: src/test/java/com/adorsys/ssi/sdjwt/sdjwtvp/SdJwtVPTest.java [hunk 4]
```diff
@@ -105,47 +108,55 @@ public void testS7_3_GhostDisclosures() {
...
         String sdJwtVPString = TestUtils.readFileAsString(getClass(), "sdjwt/s7.3-sdjwt.txt");
         SdJwtVP sdJwtVP = SdJwtVP.of(sdJwtVPString);
-        sdJwtVP.getIssuerSignedJWT().verifySignature(TestSettings.getInstance().getIssuerVerifierContext());
+
+        String jws = sdJwtVP.getIssuerSignedJWT().toJws();
+        JWSObject parse = JWSObject.parse(jws);
+        Base64URL parsedPart = parse.getParsedParts()[1];
+
+        IssuerSignedJWT issuerSignedJWT = new IssuerSignedJWT(parsedPart, testSettings.issuerSigContext.signer, testSettings.issuerSigContext.keyId, testSettings.jwsAlgorithm, "vc+sd-jwt");
+        String jws1 = issuerSignedJWT.toJws();
```
[the remark this answers is not in the document]
Comment (francis-pouatcha) [reply src/test/java/com/adorsys/ssi/sdjwt/sdjwtvp/SdJwtVPTest.java]: Fixed with [#3a6a88a](https://github.com/adorsys/sd-jwt/pull/3/commits/3a6a88a760dd18bff56bbcb487bcfada956378b1)

File: src/main/java/com/adorsys/ssi/sdjwt/vp/SdJwtVP.java [hunk 5]
```diff
@@ -90,7 +86,10 @@ public static SdJwtVP of(String sdJwtString) {
         IssuerSignedJWT issuerSignedJWT = IssuerSignedJWT.fromJws(issuerSignedJWTString);
 
         ObjectNode issuerPayload = (ObjectNode) issuerSignedJWT.getPayload();
-        String hashAlgorithm = issuerPayload.get(IssuerSignedJWT.CLAIM_NAME_SD_HASH_ALGORITHM).asText();
+        String prettyString = issuerPayload.toPrettyString();
```
[the remark this answers is not in the document]
Comment (francis-pouatcha) [reply src/main/java/com/adorsys/ssi/sdjwt/vp/SdJwtVP.java]: Fixed with [#3a6a88a](https://github.com/adorsys/sd-jwt/pull/3/commits/3a6a88a760dd18bff56bbcb487bcfada956378b1)

File: src/main/java/com/adorsys/ssi/sdjwt/SdJws.java (hunk 1 above)
Comment (francis-pouatcha) [reply src/main/java/com/adorsys/ssi/sdjwt/SdJws.java]: Fixed with [#3a6a88a](https://github.com/adorsys/sd-jwt/pull/3/commits/3a6a88a760dd18bff56bbcb487bcfada956378b1)

**Pull request merged**